// ngx-autosize/esm2015/ngx-autosize.js
import { ElementRef, HostListener, Directive, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const MAX_LOOKUP_RETRIES = 3;
class Autosize {
    /**
     * @param {?} element
     */
    constructor(element) {
        this.element = element;
        this.retries = 0;
        if (this.element.nativeElement.tagName !== 'TEXTAREA') {
            this._findNestedTextArea();
        }
        else {
            this.textAreaEl = this.element.nativeElement;
        }
    }
    /**
     * @param {?} textArea
     * @return {?}
     */
    onInput(textArea) {
        this.adjust();
    }
    /**
     * @return {?}
     */
    _findNestedTextArea() {
        this.textAreaEl = this.element.nativeElement.getElementsByTagName('TEXTAREA')[0];
        if (!this.textAreaEl) {
            if (this.retries >= MAX_LOOKUP_RETRIES) {
                console.warn('angular2-autosize: textarea not found');
            }
            else {
                this.retries++;
                setTimeout(() => {
                    this._findNestedTextArea();
                }, 100);
            }
        }
    }
    /**
     * @return {?}
     */
    ngAfterContentChecked() {
        this.adjust();
    }
    /**
     * @return {?}
     */
    adjust() {
        if (this.textAreaEl) {
            this.textAreaEl.style.overflow = 'hidden';
            this.textAreaEl.style.height = 'auto';
            this.textAreaEl.style.height = this.textAreaEl.scrollHeight + "px";
        }
    }
}
Autosize.decorators = [
    { type: Directive, args: [{
                selector: '[autosize]'
            },] },
];
/** @nocollapse */
Autosize.ctorParameters = () => [
    { type: ElementRef, },
];
Autosize.propDecorators = {
    "onInput": [{ type: HostListener, args: ['input', ['$event.target'],] },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class AutosizeModule {
    /**
     * @return {?}
     */
    static forRoot() {
        return { ngModule: AutosizeModule, providers: [] };
    }
}
AutosizeModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    Autosize
                ],
                exports: [
                    Autosize
                ]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { AutosizeModule, Autosize };
