// ngx-autosize/esm5/ngx-autosize.js
import { ElementRef, HostListener, Directive, NgModule } from '@angular/core';

var MAX_LOOKUP_RETRIES = 3;
var Autosize = (function () {
    function Autosize(element) {
        this.element = element;
        this.retries = 0;
        if (this.element.nativeElement.tagName !== 'TEXTAREA') {
            this._findNestedTextArea();
        }
        else {
            this.textAreaEl = this.element.nativeElement;
        }
    }
    Autosize.prototype.onInput = function (textArea) {
        this.adjust();
    };
    Autosize.prototype._findNestedTextArea = function () {
        var _this = this;
        this.textAreaEl = this.element.nativeElement.getElementsByTagName('TEXTAREA')[0];
        if (!this.textAreaEl) {
            if (this.retries >= MAX_LOOKUP_RETRIES) {
                console.warn('angular2-autosize: textarea not found');
            }
            else {
                this.retries++;
                setTimeout(function () {
                    _this._findNestedTextArea();
                }, 100);
            }
        }
    };
    Autosize.prototype.ngAfterContentChecked = function () {
        this.adjust();
    };
    Autosize.prototype.adjust = function () {
        if (this.textAreaEl) {
            this.textAreaEl.style.overflow = 'hidden';
            this.textAreaEl.style.height = 'auto';
            this.textAreaEl.style.height = this.textAreaEl.scrollHeight + "px";
        }
    };
    return Autosize;
}());
Autosize.decorators = [
    { type: Directive, args: [{
                selector: '[autosize]'
            },] },
];
Autosize.ctorParameters = function () { return [
    { type: ElementRef, },
]; };
Autosize.propDecorators = {
    "onInput": [{ type: HostListener, args: ['input', ['$event.target'],] },],
};
var AutosizeModule = (function () {
    function AutosizeModule() {
    }
    AutosizeModule.forRoot = function () {
        return { ngModule: AutosizeModule, providers: [] };
    };
    return AutosizeModule;
}());
AutosizeModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    Autosize
                ],
                exports: [
                    Autosize
                ]
            },] },
];

export { AutosizeModule, Autosize };
